pkg/agent/agent.go

```diff
@@ -41,38 +45,27 @@ type APIValidator func(apiID, stageName string) bool
 // ConfigChangeHandler - Callback for Config change event
 type ConfigChangeHandler func()
 
-// agentTypesMap - Agent Types map
-var agentTypesMap = map[config.AgentType]string{
-	config.DiscoveryAgent:    "discoveryagents",
-	config.TraceabilityAgent: "traceabilityagents",
-	config.GovernanceAgent:   "governanceagents",
-}
-
 type agentData struct {
-	agentResource     *apiV1.ResourceInstance
-	prevAgentResource *apiV1.ResourceInstance
+	agentResourceManager resource.Manager
 
 	apicClient       apic.Client
 	cfg              config.CentralConfig
 	agentFeaturesCfg config.AgentFeaturesConfig
 	agentCfg         interface{}
 	tokenRequester   auth.PlatformTokenGetter
-	loggerName       string
-	logLevel         string
-	logFormat        string
-	logOutput        string
-	logPath          string
-
-	apiMap                     cache.Cache
-	categoryMap                cache.Cache
+
 	teamMap                    cache.Cache
+	cacheManager               agentcache.Manager
 	apiValidator               APIValidator
 	configChangeHandler        ConfigChangeHandler
 	agentResourceChangeHandler ConfigChangeHandler
+	proxyResourceHandler       *handler.StreamWatchProxyHandler
 	isInitialized              bool
 }
 
-var agent = agentData{}
+var agent = agentData{
```

Comment:
Should the agent var here be initialized in an init func so its only created on the first import of the agent package? Does that matter?

Reply:
Done